eZ/Publish/Core/Repository/Repository.php

```diff
@@ -879,6 +744,28 @@ public function getFieldTypeService()
     }
 
     /**
+     * Get PermissionService.
+     *
+     * @return \eZ\Publish\API\Repository\PermissionService
+     */
+    public function getPermissionService()
+    {
+        if ($this->permissionService === null) {
+            $this->permissionService = new PermissionService(
+                $this,
+                $this->getUserService(),
+                $this->getRoleDomainMapper(),
+                $this->getLimitationService(),
+                $this->persistenceHandler->userHandler(),
+                $this->currentUserRef,
+                $this->currentUser
```

Comment:
I use this from `Repository`, so that protected properties there can be deprecated.
Otherwise we could move the logic from the constructor there to the PermissionService.